modes/settings_menues.h

```diff
@@ -74,42 +74,57 @@ class ShowLengthStyle {
 };
 
 template<class SPEC>
-struct ChangeBladeLengthBlade1 : public SPEC::MenuBase {
+struct ChangeBladeLengthMode : public SPEC::MenuBase {
   virtual int blade() { return menu_current_blade; }
   void mode_activate(bool onreturn) override {
     int len = prop_GetBladeLength(blade());
     int maxlen = prop_GetMaxBladeLength(blade());
-    saved_len_ = len;
     if (len == -1) len = maxlen;
-    this->pos_ = len;
+    saved_len_ = len;
+    pos_ = saved_len_;
     prop_SetBladeLength(blade(), maxlen);
     showlen_.Start(blade());
     SPEC::MenuBase::mode_activate(onreturn);
+    if (maxlen == 0) exit();
   }
   void mode_deactivate() {
     showlen_.Stop(blade());
   }
   void say() override {
-    getSL<SPEC>()->SayWhole(this->pos_);
+    getSL<SPEC>()->SayWhole(pos_);
   }
   uint16_t size() override {
     return prop_GetMaxBladeLength(blade());
   }
   void select() override {
-    prop_SetBladeLength(blade(), this->pos_);
+    prop_SetBladeLength(blade(), pos_);
     prop_SaveState();
     SPEC::MenuBase::select();
   }
   void exit() override {
+    pos_ = saved_len_;
     prop_SetBladeLength(blade(), saved_len_);
     SPEC::MenuBase::exit();
   }
 
+  void next() override {
+    pos_ = (pos_ + 1 > size()) ? 0 : pos_ + 1;
```

Comment:
These *could* all still remain as this->pos_ since inherited from MenuBase. Not sure if that's better or if this is cleaner/more appropriate since it's used pretty much exclusively here now anyway.

Reply:
pos_ should always be 0 .. size() -1

Reply:
got it.